lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -14,6 +15,28 @@ import type {
   VersionPointer,
 } from '../types';
 
+function findVersionIndex(
+  content: string,
+  depName: string,
+  version: string
+): number {
+  const contents = content.replace(regEx(/\r\n/g), '\n');
```

Comment:
Wouldn't this cause wrong `fileReplacePosition` when file contains `\r\n` ?

Reply:
nope that's why this code is here, so that \r\n is back to \n... windows fix.

Reply:
i tested this on linux and windows, on unit tests and real repository. with both formats. all good.

Reply:
you are right, ill change the `readLocalFile` from `util/fs/index.ts`

Reply:
moved it to gradle/extract.ts